torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -706,18 +735,19 @@ class PointWiseScheduler : public SchedulerEntry {
     computeHeuristics(fusion, runtime_info, data_cache);
   }
 
-  static bool canSchedule(
-      Fusion* fusion,
-      SchedulerRuntimeInfo& runtime_info,
-      HeuristicSummary* data_cache = nullptr) {
-    if (data_cache) {
-      return true;
-    }
+  static bool canScheduleCompileTime(Fusion* fusion) {
     auto red_ops = findReductionOps(fusion);
     auto welford_ops = findReductionOps<WelfordOp>(fusion);
     return red_ops.empty() && welford_ops.empty();
```

Comment:
Wonder if opt-in vs opt-out will be easier soon, we probably should rule out transpose and shift though they aren't supported yet in the parser.

Reply:
Sure I could address transpose and shift in a follow up.

Could you give more details on what `opt-in` or `opt-out` interface for schedulers would look like?